lightly/models/utils.py

```diff
@@ -484,12 +484,15 @@ def cosine_schedule(
             Target momentum value.
             
     Returns:
-        New momentum value to be used with update_momentum.
+        Cosine decay value.
 
     """
-    assert step >= 0, "Current number can't be negative"
-    assert max_steps > 0, "Total epochs must be > 0"
-    assert step <= max_steps, f"Current step is bigger than total number of steps. Found {step} > {max_steps}."
+    if step < 0:
+        raise ValueError("Current epoch number can't be negative")
+    if max_steps < 1:
+        raise ValueError("Total epochs number must be >= 1")
+    if step >= max_steps:
+        raise ValueError(f"Current step is bigger than total number of steps. Found {step} > {max_steps}.")
```

Comment:
The check and the error message do not say the same `step >= max_steps` and `step > max_steps`. 

The error message might be more helpful if it said something along the lines: `The current step must be smaller than max_steps but found step {step} and max_steps {max_steps}.`